Simulator/game_round.py

```diff
@@ -87,101 +89,71 @@ def __init__(self, game_players, pool_obj, step_func_obj):
-                        if len(alive) > 0:
-                            for other in alive:
-                                other.won_ghost(damage/len(alive))
-                    players[num].combat = True
-                    players_matched += 1
-                else:
-                    return False
+        for match in self.matchups:
+            if not match[1] == "ghost":
+                # Assigning a battle
+                if not players[match[0]] or not players[match[1]]:
+                    print(match[0], match[1])
+                players[match[0]].opponent = players[match[1]]
+                players[match[1]].opponent = players[match[0]]
+                # Fixing the time signature to see how long battles take.
+                players[match[0]].start_time = time.time_ns()
+                players[match[1]].start_time = time.time_ns()
+                config.WARLORD_WINS['blue'] = players[match[0]].win_streak
+                config.WARLORD_WINS['red'] = players[match[1]].win_streak
+
+                # Main simulation call
+                index_won, damage = champion.run(champion.champion, players[match[0]], players[match[1]],
+                                                 self.ROUND_DAMAGE[round_index][1])
+
+                # Draw
+                if index_won == 0:
+                    players[match[0]].loss_round(damage)
+                    players[match[0]].health -= damage
+                    players[match[1]].loss_round(damage)
+                    players[match[1]].health -= damage
+
+                # Blue side won
+                if index_won == 1:
+                    players[match[0]].won_round(damage)
+                    players[match[1]].loss_round(damage)
+                    players[match[1]].health -= damage
+
+                # Red side won
+                if index_won == 2:
+                    players[match[0]].loss_round(damage)
+                    players[match[0]].health -= damage
+                    players[match[1]].won_round(damage)
+                players[match[0]].combat = True
+                players[match[1]].combat = True
+
+            else:
+                players[match[0]].start_time = time.time_ns()
+                config.WARLORD_WINS['blue'] = players[match[0]].win_streak
+                config.WARLORD_WINS['red'] = players[match[2]].win_streak
+                index_won, damage = champion.run(champion.champion, players[match[0]], players[match[2]],
+                                                 self.ROUND_DAMAGE[round_index][1])
+                # if the alive player loses to a dead player, the dead player's reward is
+                # given out to all other alive players
+                alive = []
+                for other in players:
+                    if other:
+                        if other.health > 0 and other is not players[match[0]]:
+                            alive.append(other)
+                if index_won == 2 or index_won == 0:

```

Comment:
Give a positive reward to all other players equal to the negative reward here. We need to try to keep the overall game a 0 sum game.